src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.SymbolReferenceFinder.cs

```diff
@@ -44,8 +47,31 @@ private class SymbolReferenceFinder
 
                 _containingType = semanticModel.GetEnclosingNamedType(node.SpanStart, cancellationToken);
                 _containingTypeOrAssembly = _containingType ?? (ISymbol)semanticModel.Compilation.Assembly;
-                _namespacesInScope = owner.GetNamespacesInScope(semanticModel, node, cancellationToken);
                 _syntaxFacts = document.Project.LanguageServices.GetService<ISyntaxFactsService>();
+
+                _namespacesInScope = GetNamespacesInScope(cancellationToken);
+            }
+
+            private ISet<INamespaceSymbol> GetNamespacesInScope(CancellationToken cancellationToken)
+            {
+                // Add all hte namespaces brought in by imports/usings.
+                var set = _owner.GetImportNamespacesInScope(_semanticModel, _node, cancellationToken);
```

Comment:
Namespaces coming back from this call will still be MergedNamespace symbols, not compilation ones AFAICT.